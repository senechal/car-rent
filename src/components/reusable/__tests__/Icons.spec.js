import { render } from '@testing-library/react';
import icons from '../Icons';

describe('Testing Icons', () => {
  Object.keys(icons).forEach((iconName) => {
    it(`should render ${iconName} correctly`, () => {
        const Component = icons[iconName];
        const { container } = render(<Component />);
        expect(container).toMatchSnapshot();
    });
  });
});