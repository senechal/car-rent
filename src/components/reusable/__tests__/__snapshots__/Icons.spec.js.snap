// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Icons should render AcIcon correctly 1`] = `
<div>
  <span
    class="anticon rac-icon"
    role="img"
  >
    <svg
      aria-hidden="true"
      class=""
      fill="currentColor"
      focusable="false"
      height="1em"
      width="1em"
    >
      ac.svg
    </svg>
  </span>
</div>
`;

exports[`Testing Icons should render AutomaticGear correctly 1`] = `
<div>
  <span
    class="anticon rac-icon"
    role="img"
  >
    <svg
      aria-hidden="true"
      class=""
      fill="currentColor"
      focusable="false"
      height="1em"
      width="1em"
    >
      automatic.svg
    </svg>
  </span>
</div>
`;

exports[`Testing Icons should render BagIcon correctly 1`] = `
<div>
  <span
    class="anticon rac-icon"
    role="img"
  >
    <svg
      aria-hidden="true"
      class=""
      fill="currentColor"
      focusable="false"
      height="1em"
      width="1em"
    >
      bag.svg
    </svg>
  </span>
</div>
`;

exports[`Testing Icons should render CarLogo correctly 1`] = `
<div>
  <span
    class="anticon rac-car-logo"
    role="img"
  >
    <svg
      aria-hidden="true"
      class=""
      fill="currentColor"
      focusable="false"
      height="1em"
      width="1em"
    >
      car.svg
    </svg>
  </span>
</div>
`;

exports[`Testing Icons should render DoorIcon correctly 1`] = `
<div>
  <span
    class="anticon rac-icon"
    role="img"
  >
    <svg
      aria-hidden="true"
      class=""
      fill="currentColor"
      focusable="false"
      height="1em"
      width="1em"
    >
      doors.svg
    </svg>
  </span>
</div>
`;

exports[`Testing Icons should render FuelIcon correctly 1`] = `
<div>
  <span
    class="anticon rac-icon"
    role="img"
  >
    <svg
      aria-hidden="true"
      class=""
      fill="currentColor"
      focusable="false"
      height="1em"
      width="1em"
    >
      fuel.svg
    </svg>
  </span>
</div>
`;

exports[`Testing Icons should render ManualGear correctly 1`] = `
<div>
  <span
    class="anticon rac-icon"
    role="img"
  >
    <svg
      aria-hidden="true"
      class=""
      fill="currentColor"
      focusable="false"
      height="1em"
      width="1em"
    >
      manual.svg
    </svg>
  </span>
</div>
`;

exports[`Testing Icons should render SeatIcon correctly 1`] = `
<div>
  <span
    class="anticon rac-icon"
    role="img"
  >
    <svg
      aria-hidden="true"
      class=""
      fill="currentColor"
      focusable="false"
      height="1em"
      width="1em"
    >
      seats.svg
    </svg>
  </span>
</div>
`;
